### ISSUE TITLE
Save Imgur Descriptions in jpg metadata

### ISSUE THREAD

**Issue by BubiBalboa** [4pr0n/ripme#582]
Hi folks! Would it be possible to save the descriptions of the images in Imgur folders in the metadata of the pictures? Albums from r/DIY are pretty incomplete without the descriptions. Thanks!

**Comment by cyian-1756** [comment]
> Hi folks! Would it be possible to save the descriptions of the images in Imgur folders in the metadata of the pictures?

I'm pretty sure you can't add random data to jpg's and still have the jpg work in most views. Ripme can however save the descriptions as .txt files (Which it may already do for imgur if you have the `save descriptions` option checked in the config menu

**Comment by BubiBalboa** [comment]
Yeah I have seen that option (it's even checked by default). It didn't save anything though. I have done some research just now and I think the standard method would be to have an XMP file for every image.

**Comment by cyian-1756** [comment]
> Yeah I have seen that option (it's even checked by default). It didn't save anything though

Damn. Well I get on adding it in the next few days

> I have done some research just now and I think the standard method would be to have an XMP file for every image.

No idea if that will be possible without rewriting a bunch of ripmes util functions, but I'll look into it

**Comment by BubiBalboa** [comment]
Thank you so much. Another idea I read is to put out a PDF with the pictures and descriptions embedded. I don't think this is within the scope of your fine tool but I thought I mention it. :)

**Comment by metaprime** [comment]
@BubiBalboa probably out of scope, but once the text data is stored, with the image and text data, you should be able to format it however you like. As easy option would be to have a script put together a latex document which could then be compiled to a PDF with another tool.

OTOH, I would assume that print page from an imgur album would do something reasonable -- although perhaps not if the album is too large.

**Comment by rautamiekka** [comment]
At least licensing won't be a problem with XMP since it's released under BSD license (simply put: do what the hell you please).

**Comment by metaprime** [comment]
We could also fashion a simple HTML page with links to the local files and the text embedded, cutting out the bulk of formatting would keep the file small, and would avoid duplicating data like a PDF would.

**Comment by rautamiekka** [comment]
^ HTML page is a good idea, and fairly simple even if automated libs weren't used.
